server/src/main/java/io/crate/planner/operators/AbstractJoinPlan.java

```diff
@@ -48,15 +48,21 @@ public abstract class AbstractJoinPlan implements LogicalPlan {
     @Nullable
     protected final Symbol joinCondition;
     protected final JoinType joinType;
+    protected final boolean lhsIsLookup;
+    protected final boolean rhsIsLookup;
```

Comment:
An alternative could be to introduce a Lookup-Join-Operator and then check for the type of the sub-plans. However, this approach is simpler and cheaper in terms of resources.

Reply:
I'd say to keep it as is and consider introducing a Lookup-Join if in the future we add more functionality (further optimisations).